editor.js

```diff
@@ -704,17 +708,16 @@ function uploadMergeCode(append) {
                 return p;
             });
             for (var j = 0; j < oldBPCvars.length; j++) {
-                k = 0;
-                while (k < newBPCvars.length) {
+                for (var k = 0; k < newBPCvars.length; k++) {
```

Comment:
The for() is cleaner in appearance here than the while(), but I want to emphasize that it won't prevent lockups in this context.  The body was formed properly so that the k index was always incremented (as it should be), so the for only makes it less likely that a future change to the body inadvertently creates an endless loop.